packages/react-juce/src/components/Slider.tsx

```diff
@@ -0,0 +1,284 @@
+
+    value = Math.max(0.0, Math.min(1.0, valueAtDragStart + (dx + dy) * sensitivity));
+    return value;
+}
+
+const _drawLinearHorizontalSlider = (strokeColor: string,
+                                     fillColor: string) => {
+    return (ctx: CanvasRenderingContext2D,
+            width: number,
+            height: number,
+            value: number): void => {
+
+        let radius = 10;
+        const lineWidth = 3.0;
+
+        const x = 0;
+        const y = 0;
+
+        // Slider track stroke
+        ctx.lineWidth = lineWidth;
+        ctx.strokeStyle = strokeColor;
+
+        // TODO: Remove ts-ignore once we have our own CanvasContext union type
+        // @ts-ignore
+        ctx.strokeRoundedRect(x, y, width, height, radius);
+
+        // Slider track fill
+        ctx.fillStyle = fillColor;
+
+        const fillWidth = value * width;
+
+        if (fillWidth < radius) {
+            radius = fillWidth;
+        }
+
+        // TODO: Remove ts-ignore once we have our own CanvasContext union type
+        // @ts-ignore
+        ctx.fillRoundedRect(x, y, fillWidth, height, radius);
+    }
+}
+
+const _drawLinearVerticalSlider = (strokeColor: string,
+                                   fillColor: string) => {
+    return (ctx: CanvasRenderingContext2D,
+            width: number,
+            height: number,
+            value: number): void => {
+
+        let radius = 10;
+        const lineWidth = 3.0;
+
+        const x = 0;
+        const y = 0;
+
+        // Slider track stroke
+        ctx.lineWidth = lineWidth;
+        ctx.strokeStyle = strokeColor;
+
+        // TODO: Remove ts-ignore once we have our own CanvasContext union type
+        // @ts-ignore
+        ctx.strokeRoundedRect(x, y, width, height, radius);
+
+        // Slider track fill
+        ctx.fillStyle = fillColor;
+
+        const fillHeight = value * height;
+        const fillY = y + (height - fillHeight);
+
+        if (fillHeight < radius) {
+            radius = fillHeight;
+        }
+
+        // TODO: Remove ts-ignore once we have our own CanvasContext union type
+        // @ts-ignore
+        ctx.fillRoundedRect(x, fillY, width, fillHeight, radius);
+    }
+}
+
+const _drawRotarySlider = (strokeColor: string,
+                           fillColor: string) => {
+    return (ctx: CanvasRenderingContext2D,
+            width: number,
+            height: number,
+            value: number): void => {
+
+        //TODO: Finish
+        const lineWidth = 15;
+
+        const radius = Math.min(width, height) * 0.5 - lineWidth / 2;
+
+        const x = 0;
+        const y = 0;
+
+        const arcSize = 0.8;
+
+        ctx.lineWidth = lineWidth;
+        ctx.strokeStyle = strokeColor;
+        ctx.fillStyle = fillColor;
+
+        ctx.beginPath();
+        ctx.moveTo(x + radius, y);
+        ctx.arc(x, y, radius, 0, 2 * Math.PI * arcSize);
+
+        const rotateAngle = ((180 + 180 * (1 - arcSize)) * Math.PI) / 180;
```

Comment:
Not sure I follow the rotate part here. Could we just `ctx.arc` once for the rotary slider track, then `ctx.arc` again on top of it for the rotary slider fill?

Reply:
Ah yeah don't pay too much attention to this bit of drawing code :-). I copy pasted a half finished thing from elsewhere to start twiddling. Yeah totally. My issue was that it'd be nice to have an actual arc outline the same as the track fill color if possible but my software drawing skills are totaly garbage. Two arcs is the straightforward approach.